src/envoy/http/jwt_auth/jwt.cc

```diff
@@ -333,8 +333,12 @@ bool Verifier::VerifySignatureRSA(EVP_PKEY *key, const EVP_MD *md,
                                   size_t signed_data_len) {
   bssl::UniquePtr<EVP_MD_CTX> md_ctx(EVP_MD_CTX_create());
 
-  EVP_DigestVerifyInit(md_ctx.get(), nullptr, md, nullptr, key);
-  EVP_DigestVerifyUpdate(md_ctx.get(), signed_data, signed_data_len);
+  if (EVP_DigestVerifyInit(md_ctx.get(), nullptr, md, nullptr, key) != 1) {
+    return false;
```

Comment:
there should be a EVP_MD_CTX_destroy() of whatever was returned by EVP_MD_CTX_create().

Reply:
See https://github.com/google/boringssl/blob/b96d4708652ba962e147eb0733d62afea07b31b0/include/openssl/digest.h#L109

Reply:
So far,  EVP_MD_CTX_create() only allocate a piece of buffer.  bssl::UniquePtr will free the buffer. So we are OK.

If EVP_MD_CTX_create() allocate embedded multiple buffers, then we have memory leak.

Reply:
it does allocate multiple buffer, it has a ->md_data field in it, that could be allocated (c.f.: https://github.com/majek/openssl/blob/master/crypto/evp/digest.c#L206
)

Reply:
Also ->pctx is allocate as soon as the verify init is called:
https://github.com/openssl/openssl/blob/master/crypto/evp/m_sigver.c#L29

Reply:
borningssl Smart point is very small.  It knows which destroy function call.

It is defined [here](https://github.com/google/boringssl/blob/6433a91dcb153304da4ef6a3d47a98d5860252a4/include/openssl/base.h#L534)